### PR TITLE
[SYCL][Matrix] Add GNR matrix combinations to spec and runtime query

### PR DESCRIPTION
**Pull request by dkhaldi**
As a minor addition, I also added tf32 combination to PVC as part of this PR

### CODE REVIEW

File: sycl/source/detail/device_info.hpp [hunk 1]
```diff
@@ -746,6 +762,8 @@ struct get_device_info_impl<
            matrix_type::fp32, matrix_type::fp32},
           {8, 0, 0, 0, 16, 16, matrix_type::bf16, matrix_type::bf16,
            matrix_type::fp32, matrix_type::fp32},
+          {8, 0, 0, 0, 16, 8, matrix_type::tf32, matrix_type::tf32,
```
Comment (dm-vodopyanov) [inline sycl/source/detail/device_info.hpp:765]: This doesn't relate to GNR, can you please update the caption/description of PR?

File: sycl/source/detail/device_info.hpp [hunk 2]
```diff
@@ -623,6 +623,7 @@ struct get_device_info_impl<range<Dimensions>,
 // TODO: extend the macro with other CPU architectures when they will be added
 // to ext_oneapi_device_architecture
 #define INTEL_CPU_ARCHES(X) X(8, oneapi_exp_arch::intel_cpu_spr)
+#define INTEL_CPU_ARCHES(X) X(9, oneapi_exp_arch::intel_cpu_gnr)
```
Comment (dm-vodopyanov) [inline sycl/source/detail/device_info.hpp:626]:
```
#define INTEL_CPU_ARCHES(X)                  \
  X(8, oneapi_exp_arch::intel_cpu_spr)       \
  X(9, oneapi_exp_arch::intel_cpu_gnr)
```

File: sycl/source/detail/device_info.hpp (hunk 1 above)
Comment (maarquitos14) [inline sycl/source/detail/device_info.hpp:765]: I'm not very familiar with this part of the code, so maybe this is a silly question, but if the PR is meant to add GNR matrix combinations, why do we need to update PVC matrix combinations?

Comment (dkhaldi) [reply sycl/source/detail/device_info.hpp:765]: Updated description

File: sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc [hunk 3]
```diff
@@ -855,24 +855,32 @@ XMX hardware. Note that these can be returned using
...
-`matrix_type::sint32`  |  +<=+ 16 |  +<=+ 16 |  +<=+ 64
-| `matrix_type::sint8`  | `matrix_type::uint8` |
-`matrix_type::sint32`  |  +<=+ 16 |  +<=+ 16 |  +<=+ 64
-| `matrix_type::sint8`  | `matrix_type::sint8` |
-`matrix_type::sint32`  |  +<=+ 16 |  +<=+ 16 |  +<=+ 64
-|  `matrix_type::bf16`       |  `matrix_type::bf16`   |
+| A type | B type | C and D type | M | N | K | device
+.2+| `matrix_type::uint8`  .2+| `matrix_type::uint8` .2+|
+`matrix_type::sint32`  .2+|  +<=+ 16 .2+|  +<=+ 16 .2+|  +<=+ 64
+|`architecture::intel_gpu_spr`|`architecture::intel_gpu_gnr`
```
Comment (dm-vodopyanov) [inline sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc:868]: Looks like no need to divide the cell on two cells:
Currently there is 
```
===================================
| architecture::intel_gpu_spr     |
===================================
| architecture::intel_gpu_gnr     |
===================================
```
Suggesting:
```
===================================
| architecture::intel_gpu_spr     |
| architecture::intel_gpu_gnr     |
===================================
```

Comment (dkhaldi) [reply sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc]: done

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**